src/main/kotlin/Creatable.kt

```diff
@@ -0,0 +1,4 @@
+interface Creatable {
+    fun createObject() {}
+    fun createObject(archive: Archive) {}
```

Comment:
🍏 Тут происходит некое смешение логики: сначала кажется, что Creatable - это интерфейса для класса, который можно создать. Потом по названиям методов становится понятно, что этот класс умеет создавать объекты. При этом в первом методе - любые (судя по коду - только архивы), а во втором те, которые можно добавить в архив.
Я бы не делала привязку к архиву, пусть тогда в качестве контейнера выступает любой объект.
Например:
 ```suggestion
interface Creator {
    fun createObject(container: Any? = null)
```